qcodes/dataset/sqlite_base.py

```diff
@@ -1363,29 +1507,113 @@ def add_meta_data(conn: sqlite3.Connection,
             raise e
 
 
-def get_user_version(conn: sqlite3.Connection) -> int:
+def get_user_version(conn: SomeConnection) -> int:
 
     curr = atomic_transaction(conn, 'PRAGMA user_version')
     res = one(curr, 0)
     return res
 
 
-def set_user_version(conn: sqlite3.Connection, version: int) -> None:
+def set_user_version(conn: SomeConnection, version: int) -> None:
 
     atomic_transaction(conn, 'PRAGMA user_version({})'.format(version))
 
 
 def get_experiment_name_from_experiment_id(
-        conn: sqlite3.Connection, exp_id: int) -> str:
+        conn: SomeConnection, exp_id: int) -> str:
     return select_one_where(
         conn, "experiments", "name", "exp_id", exp_id)
 
 
 def get_sample_name_from_experiment_id(
-        conn: sqlite3.Connection, exp_id: int) -> str:
+        conn: SomeConnection, exp_id: int) -> str:
     return select_one_where(
         conn, "experiments", "sample_name", "exp_id", exp_id)
 
 
-def get_run_timestamp_from_run_id(conn: sqlite3.Connection, run_id: int) -> int:
+def get_run_timestamp_from_run_id(conn: SomeConnection,
+                                  run_id: int) -> float:
     return select_one_where(conn, "runs", "run_timestamp", "run_id", run_id)
+
+
+def update_GUIDs(conn: SomeConnection) -> None:
+    """
+    Update all GUIDs in this database where either the location code or the
+    work_station code is zero to use the location and work_station code from
+    the qcodesrc.json file in home. Runs where it is not true that both codes
+    are zero are skipped.
+    """
+
+    log.info('Commencing update of all GUIDs in database')
+
+    cfg = qc.Config()
+
+    try:
+        location = cfg['GUID_components']['location']
+    except KeyError:
+        log.warning('No location information found in the config. '
+                    'Can not proceed.')
+        return
+    try:
+        work_station = cfg['GUID_components']['work_station']
+    except KeyError:
+        log.warning('No work_station information found in the config. '
+                    'Can not proceed.')
+        return
+
+    if location == 0:
+        log.warning('The location is still set to the default (0). Can not '
+                    'proceed. Please configure the location before updating '
+                    'the GUIDs.')
+    if work_station == 0:
+        log.warning('The work_station is still set to the default (0). Can not'
+                    ' proceed. Please configure the location before updating '
+                    'the GUIDs.')
+
+    query = f"select MAX(run_id) from runs"
```

Comment:
do i understand correctly that since `run_id` is an integer PRIMARY KEY, there are no missing run IDs between 1 and MAX(run_id)?

Reply:
Correct. If you have missing runs, your database is fouled up. That is not possible to achieve with our API, so we don't consider it an option.